core/src/main/scala/kafka/log/OffsetMap.scala

```diff
@@ -68,90 +120,168 @@ class SkimpyOffsetMap(val memory: Int, val hashAlgorithm: String = "MD5") extend
 
   /**
    * The number of bytes of space each entry uses (the number of bytes in the hash plus an 8 byte offset)
+   * This evaluates to the number of bytes in the hash plus 8 bytes for the offset
+   * and, if applicable, another 8 bytes for non-offset compact strategy (set in the init method).
    */
-  val bytesPerEntry = hashSize + 8
-  
+  var bytesPerEntry = hashSize + longByteSize
+
   /**
    * The maximum number of entries this map can contain
    */
-  val slots: Int = memory / bytesPerEntry
+  var slots: Int = memory / bytesPerEntry
+
+  /* compact strategy */
+  private var compactionStrategy: CompactionStrategy = null
+
+  /* header key for the Strategy header to look for */
+  private var headerKey: String = ""
+
+  /**
+   * Initialize the map with the topic compact strategy
+   * @param strategy The compaction strategy
+   * @param headerKey The header key if the compaction strategy is set to header
+   * @param cleanerThreadId The clenaer thread id
+   * @param topicPartitionName The topic partition name
+   */
+  override def init(strategy: String = Defaults.CompactionStrategy, headerKey: String = "", cleanerThreadId: Int = -1, topicPartitionName: String = "") {
+    // set the log indent for the topic partition
+    this.logIdent = s"[OffsetMap-$cleanerThreadId $topicPartitionName]: "
+    info(s"Initializing OffsetMap with compaction strategy '$strategy' and header key '$headerKey'")
+
+    // Change the salt used for key hashing making all existing keys unfindable.
+    this.entries = 0
+    this.lookups = 0L
+    this.probes = 0L
+    this.lastOffset = -1L
+    Arrays.fill(bytes.array, bytes.arrayOffset, bytes.arrayOffset + bytes.limit(), 0.toByte)
+
+    this.compactionStrategy = CompactionStrategy.withName(strategy)
+    this.headerKey = headerKey.trim()
+    if (this.compactionStrategy == CompactionStrategy.HEADER && this.headerKey.isEmpty()) {
+      // fall back to offset if header key not set
```

Comment:
Is this consistent with LogConfig.scala line 349 ?

Reply:
its duplicate check here as it is already enforced part of the config validation itself... at this point, the headerKey will be present for Header strategy... will remove this check!